tsl/test/sql/compression_update_delete.sql

```diff
@@ -0,0 +1,656 @@
+
+-- get rowcount from compressed chunks where d = 3
+SELECT COUNT(*) FROM :COMPRESS_CHUNK_1 WHERE d = 3;
+SELECT COUNT(*) FROM :COMPRESS_CHUNK_2 WHERE d = 3;
+
+-- check chunk compression status after DELETE
+SELECT chunk_status,
+       chunk_name as "CHUNK_NAME"
+FROM compressed_chunk_info_view
+WHERE hypertable_name = 'sample_table' ORDER BY chunk_name;
+
+-- get rowcount from compressed chunks where device_id IS NULL
+SELECT COUNT(*) FROM :COMPRESS_CHUNK_1 WHERE device_id IS NULL;
+SELECT COUNT(*) FROM :COMPRESS_CHUNK_2 WHERE device_id IS NULL;
+
+BEGIN;
+-- report 0 row
+SELECT * FROM sample_table WHERE a = 247;
+-- delete 1 row
+UPDATE sample_table SET a = 247 WHERE device_id IS NULL;
+-- ensure rows are visible
+SELECT * FROM sample_table WHERE a = 247;
+ROLLBACK;
+
+-- report 0 rows
+SELECT COUNT(*) FROM sample_table WHERE a = 247;
+
+-- UPDATE based on NULL values in SEGMENTBY column
+UPDATE sample_table SET a = 247 WHERE device_id IS NULL;
+
+-- report 3 rows
+SELECT COUNT(*) FROM sample_table WHERE a = 247;
+
+-- ensure that not all rows are moved to staging area
+-- should have few compressed rows
+SELECT COUNT(*) FROM :COMPRESS_CHUNK_1;
+SELECT COUNT(*) FROM :COMPRESS_CHUNK_2;
+
+-- get rowcount from compressed chunks where device_id IS NULL
+-- should report 0 rows
+SELECT COUNT(*) FROM :COMPRESS_CHUNK_1 WHERE device_id IS NULL;
+SELECT COUNT(*) FROM :COMPRESS_CHUNK_2 WHERE device_id IS NULL;
+
+DROP TABLE sample_table;
+
+-- test with different physical layout
+CREATE TABLE sample_table(time timestamptz, c1 text, c2 text, c3 text);
+SELECT create_hypertable('sample_table','time');
+INSERT INTO sample_table SELECT '2000-01-01';
+ALTER TABLE sample_table DROP column c3;
+ALTER TABLE sample_table ADD column c4 text;
+INSERT INTO sample_table SELECT '2000-01-01', '1', '2', '3';
+ALTER TABLE sample_table SET (timescaledb.compress,timescaledb.compress_segmentby='c4');
+SELECT compress_chunk(show_chunks('sample_table'));
+
+BEGIN;
+-- report 1 row
+SELECT * FROM sample_table WHERE c4 IS NULL;
+-- delete 1 row
+DELETE FROM sample_table WHERE c4 IS NULL;
+-- report 0 rows
+SELECT * FROM sample_table WHERE c4 IS NULL;
+ROLLBACK;
+
+-- get FIRST compressed chunk
+SELECT ch1.schema_name|| '.' || ch1.table_name AS "COMPRESS_CHUNK_1"
+FROM _timescaledb_catalog.chunk ch1, _timescaledb_catalog.hypertable ht
+WHERE ch1.hypertable_id = ht.id AND ch1.table_name LIKE 'compress_%'
+ORDER BY ch1.id \gset
+
+-- report 2 rows
+SELECT COUNT(*) FROM :COMPRESS_CHUNK_1;
+-- report 1 row
+SELECT COUNT(*) FROM :COMPRESS_CHUNK_1 WHERE c4 IS NULL;
+
+-- report 1 row
+SELECT * FROM sample_table WHERE c4 IS NULL;
+-- delete 1 row
+DELETE FROM sample_table WHERE c4 IS NULL;
+-- report 0 row
+SELECT * FROM sample_table WHERE c4 IS NULL;
+
+-- report 1 row which ensure that only required row is moved and deleted
+SELECT COUNT(*) FROM :COMPRESS_CHUNK_1;
+-- report 0 row
+SELECT COUNT(*) FROM :COMPRESS_CHUNK_1 WHERE c4 IS NULL;
+
+DROP TABLE sample_table;
```

Comment:
Do we want to support this kind of updates on a single chunk, in the fashion of "transparent decompression"? That is, if you write `update _timescaledb_internal.chunk_1 set x = y ...`, it should also decompress the relevant parts  and so on. We do support selects like this, so it might make sense.